v1/lib/baseOutput.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * @file
+ * @version 0.1
+ * @copyright 2016 CN-Consult GmbH
+ * @author Max Späth <max.spaeth@cn-consult.eu>
+ */
+
+/**
+ * This abstract class is the base class for the output plugins.
+ * All output classes have to extend from this class.
+ */
+abstract class BaseOutput
+{
+    /**
+     * This function needs to be implemented in every plugin which extends from this class
```

Comment:
"." at the end, this is necessary in our diloc code, for generating a documentation with doxygen. So it's better when you learn it right from the beginning.